### PR TITLE
[tests] Add In-Process Aggregator (Start Only)

### PR DESCRIPTION
**Pull request by fishie9**
**What this PR does / why we need it**:

This PR adds resources.Aggregator interface with a few basic functions.
It also adds an in-process implementation of the new interface which will start in process. 

More PRs to follow:
- Support of starting multiple aggregator instances.
- Implementation of the functions in resources.Aggregator.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/cmd/services/m3aggregator/config/config.go [hunk 1]
```diff
@@ -26,13 +26,32 @@ import (
 	"github.com/m3db/m3/src/x/log"
 )
 
+var (
+	defaultMetricsSanitization        = instrument.PrometheusMetricSanitization
```
Comment (nbroyles) [inline src/cmd/services/m3aggregator/config/config.go:30]: Where do these defaults come from?

File: src/cmd/services/m3aggregator/config/aggregator.go [hunk 2]
```diff
@@ -89,7 +89,7 @@ type AggregatorConfiguration struct {
 	GaugePrefix *string `yaml:"gaugePrefix"`
 
 	// Stream configuration for computing quantiles.
-	Stream streamConfiguration `yaml:"stream"`
+	Stream *streamConfiguration `yaml:"stream"`
```
Comment (nbroyles) [inline src/cmd/services/m3aggregator/config/aggregator.go:92]: Why make this nillable?

File: src/integration/resources/inprocess/aggregator_test.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+	"github.com/m3db/m3/src/cluster/generated/proto/placementpb"
+	"github.com/m3db/m3/src/integration/resources"
+	"github.com/m3db/m3/src/msg/generated/proto/topicpb"
+	"github.com/m3db/m3/src/query/generated/proto/admin"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewAggregator(t *testing.T) {
```
Comment (nbroyles) [inline src/integration/resources/inprocess/aggregator_test.go:38]: Can we add a test that restarts the agg in the same test? I have a similar update on one of my upcoming PRs, but it's a good way to ensure agg is always restartable in process.

Something like:
```go
func TestNewAggregator(t *testing.T) {
  // setup

  agg, err := NewAggregator(defaultAggregatorConfig, AggregatorOptions{})
  require.NoError(t, err)
  require.NoError(t, agg.Close())

  agg, err = NewAggregator(defaultAggregatorConfig, AggregatorOptions{})
  require.NoError(t, err)
  require.NoError(t, agg.Close())
}
```

File: src/integration/resources/inprocess/coordinator_test.go [hunk 4]
```diff
@@ -279,3 +279,37 @@ db:
     filePathPrefix: "*"
   writeNewSeriesAsync: false
 `
+
+const aggregatorCoordConfig = `
```
Comment (nbroyles) [inline src/integration/resources/inprocess/coordinator_test.go:283]: Let's move this to the `aggregator_test.go`

File: src/integration/resources/inprocess/aggregator_test.go (hunk 3 above)
Comment (fishie9) [reply src/integration/resources/inprocess/aggregator_test.go:38]: Done. But this test is gonna take long.

Comment (fishie9) [reply src/integration/resources/inprocess/aggregator_test.go:38]: Done. This test is taking long though.

File: src/cmd/services/m3aggregator/config/aggregator.go (hunk 2 above)
Comment (fishie9) [reply src/cmd/services/m3aggregator/config/aggregator.go:92]: I was trying to cut the unused part in the test config yaml.
Just changed it back and added a minimal stream configuration in the test.

File: src/cmd/services/m3aggregator/config/config.go (hunk 1 above)
Comment (fishie9) [reply src/cmd/services/m3aggregator/config/config.go]: Copied from the dbnode test. 
I reverted this change and added the metrics config in the test yaml again.
I can work on the default values later in separate PRs.

File: src/integration/resources/inprocess/aggregator_test.go [hunk 5]
```diff
@@ -0,0 +1,278 @@
+//go:build integration_v2
```
Comment (nbroyles) [inline src/integration/resources/inprocess/aggregator_test.go:1]: Let's get rid of one of these. Only need one tag here.

**Review by nbroyles — APPROVED** [review]

**Pull request merged**